### PR TITLE
Start puma in `single-mode` by default.

### PR DESCRIPTION
**Pull request by parndt**
We want to ensure that Hanami runs using the lowest memory footprint by
default.

### CODE REVIEW

File: lib/hanami/cli/generators/gem/app/puma.erb [hunk 1]
```diff
@@ -6,7 +6,7 @@ threads min_threads_count, max_threads_count
 
 port        ENV.fetch("<%= Hanami::Port::ENV_VAR %>", <%= Hanami::Port::DEFAULT %>)
 environment ENV.fetch("HANAMI_ENV", "development")
-workers     ENV.fetch("HANAMI_WEB_CONCURRENCY", 2)
+workers     ENV.fetch("HANAMI_WEB_CONCURRENCY", 1)
```
Comment (parndt) [inline lib/hanami/cli/generators/gem/app/puma.erb:9]:
```suggestion
workers     ENV.fetch("HANAMI_WEB_CONCURRENCY", 0)
```

File: spec/unit/hanami/cli/commands/gem/new_spec.rb [hunk 2]
```diff
@@ -183,7 +183,7 @@ class Routes < Hanami::Routes
 
         port        ENV.fetch("HANAMI_PORT", 2300)
         environment ENV.fetch("HANAMI_ENV", "development")
-        workers     ENV.fetch("HANAMI_WEB_CONCURRENCY", 2)
+        workers     ENV.fetch("HANAMI_WEB_CONCURRENCY", 1)
```
Comment (parndt) [inline spec/unit/hanami/cli/commands/gem/new_spec.rb:186]:
```suggestion
        workers     ENV.fetch("HANAMI_WEB_CONCURRENCY", 0)
```

**Review by jodosha — COMMENTED** [review]
@parndt What's the memory usage reduction?

**Comment by parndt** [comment]
@jodosha Unscientifically:

Before:

image

After:

image

File: lib/hanami/cli/generators/gem/app/puma.erb [hunk 3]
```diff
@@ -6,10 +6,12 @@ threads min_threads_count, max_threads_count
 
 port        ENV.fetch("<%= Hanami::Port::ENV_VAR %>", <%= Hanami::Port::DEFAULT %>)
 environment ENV.fetch("HANAMI_ENV", "development")
-workers     ENV.fetch("HANAMI_WEB_CONCURRENCY", 2)
+workers     ENV.fetch("HANAMI_WEB_CONCURRENCY", 0)
 
-on_worker_boot do
-  Hanami.shutdown
+if ENV.fetch("HANAMI_WEB_CONCURRENCY", 0) > 0
+  on_worker_boot do
```
Comment (jodosha) [inline lib/hanami/cli/generators/gem/app/puma.erb:12]: @parndt Why this change? `Hanami.shutdown` is meant to free resources like database connection.

Comment (parndt) [reply lib/hanami/cli/generators/gem/app/puma.erb:12]: @jodosha basically because it printed a warning telling me to: 
>Warning: You specified code to run in a `on_worker_boot` block, but Puma is not configured to run in cluster mode (worker count > 0 ), so your `on_worker_boot` block did not run
>Puma starting in single mode...

**Comment by timriley** [comment]
(Just for the record, Phil worked on this because I listed it here: https://trello.com/c/dBn7WI2n/22-only-start-1-puma-worker-in-dev-server)

**Comment by timriley** [comment]
I'm going to merge this now, I think it's fine as an interim step. I've made another Trello card to revisit the overall Puma config before our next release: https://trello.com/c/DOxODQXw/51-revisit-default-puma-config

**Comment by timriley** [comment]
Thanks again, @parndt!
